### ISSUE TITLE
Using sqlite3orm w/ regular Node.js

### ISSUE THREAD

**Issue by robogeek** [gms1/HomeOfThings#17]
Your documentation says sqlite3orm works with JavaScript or TypeScript.

With plain Node.js (both v20 and v22) I created a file containing the following:

```js

import {
    field,
    FieldOpts,
    fk,
    id,
    index,
    table,
    TableOpts,
    SqlDatabase,
    schema,
    BaseDAO
} from 'sqlite3orm';


// import sqlite3 from 'sqlite3';
// const db = new sqlite3.Database(':memory:');

const db = new SqlDatabase();
// opens a memory database in shared mode
await db.open('file:sqlite3orm?mode=memory&cache=shared');
// get the user_version from the database:
const userVersion = await db.getUserVersion();

@table({ name: 'FILES' })
class File {

    @id({
        name: 'file_id',
        dbtype: 'INTEGER PRIMARY KEY AUTOINCREMENT'
    })
    userId /* !: number */;

    @field({
        name: 'name', dbtype: 'TEXT'
    })
    name;

    @field({
        name: 'vpath', dbtype: 'TEXT'
    })
    vpath;
    
    @field({
        name: 'info', dbtype: 'TEXT', isJson: true
    })
    info /* : any */;
}

await schema().createTable(db, 'FILES');
```

When attempting to run the file -- the source file is named with the `.mjs` extension, meaning it is an ESM module - I get this error.

```shell
file:///home/david/Projects/akasharender/akasharender/cache/file-cache-sqlite.mjs:34
@table({ name: 'FILES' })
^

SyntaxError: Invalid or unexpected token
    at compileSourceTextModule (node:internal/modules/esm/utils:340:16)
    at ModuleLoader.moduleStrategy (node:internal/modules/esm/translators:102:18)
    at #translate (node:internal/modules/esm/loader:433:12)
    at ModuleLoader.loadAndTranslate (node:internal/modules/esm/loader:480:27)
    at async ModuleJob._link (node:internal/modules/esm/module_job:112:19)

Node.js v22.11.0
```

It's pointing at the `@` in the source code as the error.  All the results I get searching for this error are from a previous era when we used Babel to implement decorators.  The same error happens on Node.js 20.

I tried looking for a polyfill or something which would add decorator support to plain JavaScript.  Nothing.

Do I need to switch my project to TypeScript?

**Comment by gms1** [comment]
Hi @robogeek

what previous era do you mean? as far as i know decorators are still a proposal, aren't they?

With e.g. babel it should be possible to run this using JavaScript, but I have to admit that I have no experience with this and failed miserably in a 30 minute attempt 

Then i tested your script using ts-node and got it working in no time

index.ts:
```TypeScript
import {
    field,
    id,
    table,
    SqlDatabase,
    schema,
} from 'sqlite3orm';


// import sqlite3 from 'sqlite3';
// const db = new sqlite3.Database(':memory:');


@table({ name: 'FILES', autoIncrement: true  })
class File {

    @id({
        name: 'file_id',
        dbtype: 'INTEGER NOT NULL'
    })
    userId /* !: number */;

    @field({
        name: 'name', dbtype: 'TEXT'
    })
    name;

    @field({
        name: 'vpath', dbtype: 'TEXT'
    })
    vpath;

    @field({
        name: 'info', dbtype: 'TEXT', isJson: true
    })
    info /* : any */;
}


(async () => {
  const db = new SqlDatabase();
  // opens a memory database in shared mode
  await db.open('files.db');
  // get the user_version from the database:
  const userVersion = await db.getUserVersion();
  await schema().createTable(db, 'FILES');
})();
```

tsconfig.json
```JSON
{
  "module": "nodenext",
  "target": "es2022", 
  "types": ["node"],
  "lib": ["es2015"],
  "files": [],
  "include": [],
  "compilerOptions": {
    "typeRoots": ["../node_modules/@types", "typings"],
    "esModuleInterop": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true
  }
}
```

```bash
$ npm i sqlite3orm ts-node typescript @types/node
$ npx ts-node-script index.ts

$ sqlite3 files.db .schema
CREATE TABLE IF NOT EXISTS "FILES" (
  `file_id` INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
  `name` TEXT,
  `vpath` TEXT,
  `info` TEXT
);
CREATE TABLE sqlite_sequence(name,seq);

```

hope this helps

**Comment by gms1** [comment]
otherwise I can only agree with that:

https://github.com/vlio20/utils-decorators/issues/174#issuecomment-2461424016
quote: there is no support of decorators in pure js yet.

**Comment by robogeek** [comment]
In the README you say:

> This module allows you to map your model, written in JavaScript or TypeScript

Which I thought meant one could write a .js or .cjs file, no TypeScript compiler in sight, and use this package.  That didn't work out.

The "bug" (if any) here is that the README gave me a misleading conclusion.  Perhaps that could be cleared up.

As for what I meant by the "new era" of Decorators, look at the TypeScript documentation:  https://www.typescriptlang.org/docs/handbook/decorators.html

At the top it says that page describes the old Stage 2 Decorators implementation, then it refers to a blog post describing new Decorator support in the latest TypeScript.

Further - https://2ality.com/2022/10/javascript-decorators.html - is Dr. Axel Rauschmayer discussing the 2022-23 Decorators, which are now Stage 3, and how they work.  But it's also clear now that I reread that, the current status is a Babel plugin.

FWIW the descriptors described there are different from the TypeScript decorators.  I'd written an in-depth article series a couple years ago about TypeScript decorators -- what Dr. Axel describes cannot do some of the things I worked out for TypeScript decorators.

It appears I was jumping ahead of the game a little bit.

As a result, to start using your library in my application (https://akashscms.com) I had to convert it over to TypeScript.
